logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java

```diff
@@ -0,0 +1,198 @@
+package org.logstash.execution;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jruby.Ruby;
+import org.jruby.RubyBasicObject;
+import org.jruby.RubyClass;
+import org.jruby.RubyThread;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
+
+@JRubyClass(name = "ShutdownWatcher")
+public final class ShutdownWatcherExt extends RubyBasicObject {
+
+    private static final Logger LOGGER = LogManager.getLogger(ShutdownWatcherExt.class);
+
+    private static final AtomicBoolean unsafeShutdown = new AtomicBoolean(false);
+
+    private final List<IRubyObject> reports = new ArrayList<>();
+
+    private final AtomicInteger attemptsCount = new AtomicInteger(0);
+
+    private final AtomicBoolean running = new AtomicBoolean(false);
+
+    private long cyclePeriod = TimeUnit.SECONDS.toMillis(5L);
```

Comment:
Did you intend to change this from the 1sec interval in the Ruby implementation?

Reply:
Also, won't this definition clash with the later call to `TimeUnit.SECONDS.sleep()`?

Reply:
@danhermann whoops thanks for catching, fixing :)